core/deployment/src/main/java/io/quarkus/deployment/ApplicationArchiveImpl.java

```diff
@@ -31,24 +36,30 @@ public IndexView getIndex() {
     }
 
     @Override
+    @Deprecated
     public Path getArchiveRoot() {
-        return archiveRoot;
+        return rootDirs.iterator().next();
```

Comment:
```suggestion
        return rootDirs.getSinglePath();
```

Reply:
Not yet, single path is assertive, it'll throw an exception in case there are more. In our current impl (not this PR but actually current codebase) we actually pick the first path in a few places.